### PR TITLE
fix(web): respect minify option

### PR DESCRIPTION
**Pull request by jaulz**
Currently, the minify option is not respected when you want to start Expo like this:
```
expo start --no-minify --no-dev --web
```

### CODE REVIEW

**Comment by jaulz** [comment]
Not sure if I got everything since I am not 100% familiar with the code base and it's difficult catch everything 😊

File: packages/webpack-config/webpack/webpack.prod.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ module.exports = async function(env = {}, argv) {
         locations.absolute(info.absoluteResourcePath).replace(/\\/g, '/'),
     },
     optimization: {
-      minimize: true,
+      minimize: env.minify,
```
Comment (EvanBacon) [inline packages/webpack-config/webpack/webpack.prod.js]: Ensure the default value is `true` if it's undefined.

Comment (EvanBacon) [inline packages/webpack-config/webpack/webpack.prod.js]: This isn't minifying in dev mode.

**Comment by jaulz** [comment]
@EvanBacon okay, I just added your feedback. Wouldn't that also mean that we could drop most of the stuff from `applyEnvironmentVariables`?

**Pull request closed without merging**